### ISSUE TITLE
bug using tfenv with cdktf

### ISSUE THREAD

**Issue by stephengroat** [tfutils/tfenv#348]
have a strange bug where `tfenv` fails from within `cdktf`, but does not fail when `TFENV_DEBUG=3` is set

not sure how to debug or determine what's wrong

```
$ cdktf synth
######################################################################## 100.0%
######################################################################## 100.0%
######################################################################## 100.0%
######################################################################## 100.0%
[2022-07-16T18:38:04.540] [ERROR] default - /usr/src/app/.tfenv/lib/tfenv-exec.sh: line 43: /usr/src/app/.tfenv/versions/1.2.4/terraform: Permission denied
/usr/src/app/.tfenv/lib/tfenv-exec.sh: line 43: exec: /usr/src/app/.tfenv/versions/1.2.4/terraform: cannot execute: Permission denied
Terraform CLI not present - Please install a current version https://learn.hashicorp.com/terraform/getting-started/install.html
```

**Comment by SandipGorde-TomTom** [comment]
Hi, Even I am also getting similar error. Can you share how can we fix the issue ?

**Comment by davidjeddy** [comment]
Getting the same error here. Triggering tfenv in a Jenkins pipeline in parallel steps. From a quick view it looks like two processes are trying to execute `tfenv-exec.sh` at the same time. In the case of CDK, @SandipGorde-TomTom and @stephengroat can you validate if your processes are also running in parallel?

**Comment by jamesdehart** [comment]
Check out https://github.com/tfutils/tfenv/issues/398#issue-1734959458  This helped me resolve my issue.

**Comment by djakielski** [comment]
Hey @jamesdehart could you share how you solved this with cdktf?

**Comment by davidjeddy** [comment]
@jamesdehart djakielski, @jamesdehart details in #398 tfenv should be invoked with `require_serial: true` configuration during the first invokation. This will prevent the error from being thrown.

**Comment by Zordrak** [comment]
Closing due to age — this has been stale since 2022 with no follow-up. CDKTF compatibility issues are typically caused by TFENV_DIR or PATH configuration rather than tfenv itself. If anyone is still experiencing this with a current tfenv and CDKTF version, please open a new issue with reproduction steps and environment details.
